Specs/DateTime/English/DateTimeModel.json

```diff
@@ -17680,5 +17680,30 @@
         }
       }
     ]
+  },
+  {
+    "Input": "dinner with mom weekend.",
+    "Context": {
+      "ReferenceDateTime": "2016-11-11T00:00:00"
+    },
+    "NotSupported": "javascript, python",
+    "Results": [
+      {
+        "Text": "weekend",
+        "Start": 16,
+        "End": 22,
+        "TypeName": "datetimeV2.daterange",
+        "Resolution": {
+          "values": [
```

Comment:
As this is not explicit. Shouldn't we return a past and a future resolution like in other places? Why is this not happening?
Can you add cases for sentences with the "the" article too. Or other cases?
BTW, "any weekend" and "all weekends" should be supported as Set. Can we add this in the same PR?

Reply:
Past and future resolutions are returned in DatePeriodParser (with identical values) but not in DateTimeModel for cases like "(this|the|next) (week|weekend|month|year)".

Reply:
added test case "the weekend", and support for 'any' + [Unit] in Set. 
Support for 'all' + [Unit] will come later because additional work is needed to check that the unit is plural.